### PR TITLE
Feature/bb27Add a method to the controller that looks up a Wallets list by part of some Wallet name

### PR DESCRIPTION
**Pull request by Krisu93**

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletController.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import pl.byczazagroda.trackexpensesappbackend.service.WalletService;
 
 import javax.validation.Valid;
-import javax.validation.constraints.Min;
+import javax.validation.constraints.*;
```
Comment (MateuszBednarczyk) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletController.java:14]: https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletService.java [hunk 2]
```diff
@@ -7,7 +7,7 @@
 import pl.byczazagroda.trackexpensesappbackend.dto.WalletDTO;
 
 import javax.validation.Valid;
-import javax.validation.constraints.Min;
+import javax.validation.constraints.*;
```
Comment (MateuszBednarczyk) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletService.java:10]: You shouldn't import the whole package ``javax.validation.constraints``

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Request changes.
When you will fix the code, please do update your local working branch with commits from the remote. So you will be able to resolve conflicts (if exist) locally.

**Review by dawciobiel — APPROVED** [review]
Looks good for me.

**Comment by dawciobiel** [comment]
@Krisu93 Please do update the project with remove `develop`, merge it and then again commit with push to resolve conflicts locally.

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java [hunk 3]
```diff
@@ -72,4 +72,9 @@ public void deleteWalletById(Long id) {
             throw new ResourceNotDeletedException("Value does not exist in the database, please change your request");
         }
     }
+
+    @Override
+    public List<WalletDTO> getWalletsByName(String name) {
+        return getWallets().stream().filter(walletDTO -> walletDTO.name().contains(name)).toList();
```
[the remark this answers is not in the document]
Comment (dawciobiel) [reply src/main/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImpl.java:78]: For a sane user, there will only be 1-few wallets in the app.
However, your point is correct.

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java [hunk 4]
```diff
@@ -365,4 +365,27 @@ void shouldThrowAnExceptionWhenWalletIdEqualsZero() throws Exception {
         //then
         result.andExpect(status().isNoContent());
     }
+
+    @Test
+    @DisplayName("Should return information about all wallets funded by part of a wallet name and http ok status")
```
[the remark this answers is not in the document]
Comment (dawciobiel) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java:370]: Should be:
_It should return information about all wallets indicated by a part of the name and `HttpStatus.OK`_

File: src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImplTest.java [hunk 5]
```diff
@@ -259,6 +262,27 @@ void shouldThrowAnExceptionWhenWalletWithIdDoesNotExist() throws Exception {
...
 
+    @Test
+    @DisplayName("Should return list of WalletDTO by name with proper size")
+    void shouldReturnListOfWalletDTOByNameWithProperSize() {
+        // given
+        String walletNameSearched = "Family";
+        List<Wallet> walletList = createListOfWalletsByName("Family wallet", "Common Wallet", "Smith Family Wallet");
+        List<WalletDTO> walletListDTO = walletList.stream().map((Wallet x) -> new WalletDTO(x.getId(), x.getName(), x.getCreationDate())).toList();
+        given(walletRepository.findAll()).willReturn(walletList);
+        walletList.forEach(wallet -> given(walletModelMapper.mapWalletEntityToWalletDTO(wallet)).willReturn(walletListDTO.stream().filter(walletDTO -> Objects.equals(wallet.getName(), walletDTO.name())).findAny().orElse(null)));
```
[the remark this answers is not in the document]
Comment (dawciobiel) [reply src/test/java/pl/byczazagroda/trackexpensesappbackend/service/WalletServiceImplTest.java:273]: Create `Wallet` entity list.
On the base of it create `WalletDTO` list.
For `given(walletRepository.findAll())` return list of `Wallet`.
Then find on the result list is any `WalletDTO` that is equals with given `Wallet`.
	To do it compare fields: [`wallet.name`, `walletDTO.name`.]
	
Line #273 is 221 characters long!
Refactoring needed.

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required.

**Review by dawciobiel — APPROVED** [review]
Looks good for me

**Review by purbanski-deftcode — APPROVED** [review]

**Review by dawciobiel — APPROVED** [review]
Looks good for me

**Review by DawidStuzynski — APPROVED** [review]

**Review by DawidStuzynski — APPROVED** [review]
Looks good for me

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletController.java [hunk 6]
```diff
@@ -20,50 +22,75 @@
...
         return new ResponseEntity<>(walletDTO, headers, HttpStatus.CREATED);
     }
 
     @GetMapping()
-    ResponseEntity<List<WalletDTO>> getWallets() {
+    ResponseEntity<List<WalletDTO>> getAll() {
 
-        List<WalletDTO> walletsDTO = walletService.getWallets();
-        HttpHeaders headers =  new HttpHeaders();
+        List<WalletDTO> walletsDTO = walletService.getAll();
```
Comment (MateuszBednarczyk) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletController.java:52]: I would like to suggest `walletDTOs`.

**Review by MateuszBednarczyk — APPROVED** [review]

**Review by dawciobiel — APPROVED** [review]
Look good for me

**Pull request merged**